### PR TITLE
test(gov/xgns): improve test coverage with table-driven pattern

### PR DESCRIPTION
**Pull request by junghoon-vans**
## Summary
This PR significantly improves the test coverage and code quality of the xGNS contract by converting all tests to follow the table-driven testing pattern.

## Changes
- **Refactored test structure**: Converted all test functions to table-driven tests with given-when-then pattern
- **Enhanced test coverage**: Added comprehensive test cases for all functions including:
  - Success cases (normal operations, zero amounts)
  - Failure cases (negative amounts, insufficient balance, unauthorized callers)
  - Edge cases (VotingSupply with various balance combinations)
  - Halt state tests for all mint/burn operations
- **Improved test isolation**: Added cleanup helper function to ensure proper state reset between tests
- **Code consistency**: Renamed test functions to follow `TestXgns_{FunctionName}` naming convention
- **Test organization**: Consolidated duplicate tests and removed redundant test code

## Test Functions Covered
- `TestXgns_TotalSupply` - Verify total supply calculations
- `TestXgns_Mint` - Test minting by gov staker with various scenarios
- `TestXgns_MintByLaunchPad` - Test minting by launchpad contract
- `TestXgns_Burn` - Test burning by gov staker
- `TestXgns_BurnByLaunchPad` - Test burning by launchpad contract
- `TestXgns_VotingSupply` - Test voting supply calculation excluding launchpad balance
- `TestXgns_BalanceOf` - Test balance queries
- `TestXgns_Render` - Test render function with different paths
- `TestXgns_Halted` - Test all operations under halted state

### CODE REVIEW

File: contract/r/gnoswap/gov/xgns/xgns_test.gno [hunk 1]
```diff
@@ -16,83 +17,603 @@ var (
...
 func getContractAddress(roleName string) address {
 	addr, exists := access.GetAddress(roleName)
 	if !exists {
 		return rbac.DefaultRoleAddresses[prbac.SystemRole(roleName)]
 	}
-
 	return addr
 }
+
+func cleanup() {
```
[the remark this answers is not in the document]
Comment (notJoon) [reply contract/r/gnoswap/gov/xgns/xgns_test.gno]: It would be better to add `t.Helper()` at the top of the function to mark it as a test helper function.

**Review by notJoon — APPROVED** [review]
LGTM 👍 

When validating abort or panic messages, you can conveniently check long error messages by using `AbortsContains` or `PanicsContains` in `uassert` package. Give it a try next time.

**Review by jinoosss — APPROVED** [review]
